orangecontrib/text/widgets/owcorpusviewer.py

```diff
@@ -304,7 +304,8 @@ def update_info(self):
     def commit(self):
         if self.corpus is not None:
             matched = self.corpus[self.output_mask]
-            unmatched_mask = [i for i in range(len(self.corpus)) if i not in self.output_mask]
+            output_mask = set(self.output_mask)
+            unmatched_mask = [i for i in range(len(self.corpus)) if i not in output_mask]
```

Comment:
You're right, it's not. Currently `QStandardItemModel.appendRow` — adding documents to the scrollable list view — takes most of the time.